client/src/app/app-routing.module.ts

```diff
@@ -1,9 +1,24 @@
 import { NgModule } from '@angular/core';
 import { RouterModule, Routes } from '@angular/router';
 
-import { EmployeeComponent } from './employee/employee.component';
-
-const routes: Routes = [{ path: 'employees', component: EmployeeComponent }];
+const routes: Routes = [
+  {
+    path: 'employees',
+    loadChildren: () =>
+      import('./employee/employee.module').then((m) => m.EmployeeModule),
```

Comment:
This syntax ensures that the token `EmployeeModule` is not discovered by the compiler as being a dependency of this ^ (`App.Module.ts`) module, which means the webpack bundler can separate out the entire contents of `EmployeeModule`, and everything it contains, into a separate file/chunk.

This results in the initial app payload not including any `EmployeeModule` stuff and it's only downloaded when the user routes to `/employees`.

This applies as well to the other lazy loaded modules in this route definition.

https://ultimatecourses.com/blog/lazy-load-angular-modules

Reply:
Good to know, cool to see this implemented so I can follow how this all works. Interesting to see the changes you made in `app-routing.module`  and how that lines up with the `links` array in the `appcomponent`.

Reply:
Yes, the route / path management can be centralized. There is even a library to help give strong typing to routes.